src/OrchardCore.Modules/OrchardCore.Users/Controllers/ExternalAuthenticationsController.cs

```diff
@@ -215,14 +219,16 @@ public async Task<IActionResult> ExternalLoginCallback(string returnUrl = null,
             }, ModelState.AddModelError);
 
             // If the registration was successful we can link the external provider and redirect the user.
-            if (iUser == null)
+            if (iUser == null || !ModelState.IsValid)
             {
+                _logger.LogError("Unable to create internal account and link it to the external user.");
```

Comment:
I think all of these log messages would be more useful if they were to include the username as well. Otherwise, unless you're testing this right there, you won't know the context.

Reply:
We can't log private info like username or id. The logs are helpful to understand what is happening during external login.

Reply:
Why would the username or ID be private info in this context? These logs will be inspected by developers of the platform, who already have access to the source code, and most possibly in some form to the DB as well.

Alone, this log message only communicates that something went wrong with account creation for somebody. This isn't enough information to troubleshoot the issue, unless login is completely broken.

Reply:
@sebastienros onces said event user is is private info and should not be logged. Here maybe it not an issue to log username since the user does not actually exists in the database.

Reply:
Yeah but if a user asks for their data to be removed, its not simply to remove their info from the logs which puts you at the risk of not being complaint.

Reply:
I would say it's fine to mask a value. For example with a username, keep the first/last char and output something like `s******s` (with a fixed number of stars)` and an email keeping the domain part. Note that this could be a service later as this needs more design than just through a generic masking service. There are utilities in the dotnet extensions repo for that.

Reply:
@Piedone we can "in another PR" start logging masked private data. That PR will provide the masking service or static extensions for making different things like Id, Username, email... but I need to finish this PR to address immediate problem.

Reply:
I wouldn't want to do a masking service here. Good point about the necessity to remove user accounts. However, user IDs don't fall under this and thus can be logged.

Reply:
Let's move this good convo into a dedicated issue please: https://github.com/OrchardCMS/OrchardCore/issues/17075